cmd/podman/common/completion.go

```diff
@@ -1522,6 +1523,29 @@ func AutocompleteCgroupManager(cmd *cobra.Command, args []string, toComplete str
 	return types, cobra.ShellCompDirectiveNoFileComp
 }
 
+// AutocompleteContainersConfModules- Autocomplete containers.conf modules.
+func AutocompleteContainersConfModules(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	dirs, err := config.ModuleDirectories()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	var modules []string
+	for _, d := range dirs {
+		_ = filepath.Walk(d,
+			func(path string, f os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+				if path == d || f.IsDir() {
+					return nil
+				}
+				modules = append(modules, strings.TrimPrefix(path, d+"/"))
```

Comment:
I suggest you store the module name and before appending to the results make `if strings.HasPrefix(path, toComplete)` to make sure to only return valid results for the current value. While shells will filter this anyway it is required together with ShellCompDirectiveDefault because if no results are returned it would do the standard file path completion in your shell.

Also use filepath.Separator just to future proof this if we ever enable it on windows